lib/net.js

```diff
@@ -745,38 +747,14 @@ Socket.prototype._writeGeneric = function(writev, data, encoding, cb) {
     return false;
   }
 
-  var req = new WriteWrap();
-  req.handle = this._handle;
-  req.oncomplete = afterWrite;
-  req.async = false;
-  var err;
-
-  if (writev) {
-    var allBuffers = data.allBuffers;
-    var chunks;
-    var i;
-    if (allBuffers) {
-      chunks = data;
-      for (i = 0; i < data.length; i++)
-        data[i] = data[i].chunk;
-    } else {
-      chunks = new Array(data.length << 1);
-      for (i = 0; i < data.length; i++) {
-        var entry = data[i];
-        chunks[i * 2] = entry.chunk;
-        chunks[i * 2 + 1] = entry.encoding;
-      }
-    }
-    err = this._handle.writev(req, chunks, allBuffers);
-
-    // Retain chunks
-    if (err === 0) req._chunks = chunks;
-  } else {
-    err = createWriteReq(req, this._handle, data, encoding);
-  }
+  let ret;
+  const req = new WriteWrap();
+  const handle = this._handle;
+  this.createWriteWrap(req, null, handle, afterWrite);
+  if (writev) ret = this.writevGeneric(req, data, cb);
+  else ret = this.writeGeneric(req, data, encoding, cb);
```

Comment:
Style nit: Can you put the bodies on their own lines? That’s more common style inside Node core, I think